api/src/components/link/service.ts

```diff
@@ -23,12 +38,34 @@ export const doesLinkExist = async (fromPublicationId: string, toPublicationId:
 };
 
 export const deleteLink = async (id: string) => {
+    const link = await client.prisma.links.findFirst({
+        where: {
+            id
+        },
+        select: {
+            publicationFromRef: {
+                select: {
+                    id: true
+                }
+            }
+        }
+    });
+    const latestVersionFrom = await client.prisma.publicationVersion.findFirst({
```

Comment:
The same as above, we already have a service which returns the latest version for a given publication id